Updated metadata documentation part 3 - probabilities (#1866)
metoppv/improver: doc/source/metadata/Statistics.rst

.. _stat-section:

> Statistical Processsing

Author: Corrected.

=======================

.. contents:: Contents
    :depth: 3

Context
-------

> As IMPROVER both processes forecast data that has been

Reviewer: ```suggestion
IMPROVER both processes forecast data that has been
```

Author: Much better!

> statistically process and supports a range of statistical processing steps.

Reviewer: ```suggestion
statistically processed and supports a range of statistical processing steps.
```
Missed this one

Author: So did I - good spot.

> statistically process end supports a range of statistical processing steps.

Reviewer: ```suggestion
statistically processed and supports a range of statistical processing steps.
```

Author: Good spot!

the statistical processing has been calculated. 
If more than one cell method is to be indicated,
they should be arranged in the order they were applied.
The left-most operation is assumed to have been applied first. 

It is possible to indicate more precisely how the cell method was applied
by including extra information in parentheses after the method.
This information includes standardized and non-standardized parts.
The only standardized option is ``interval``,
used to provide the typical interval between the original data values
to which the method was applied,
in the situation where the present data values are statistically representative 
of original data values which had a finer spacing.

.. warning::

    It is important to understand that the following example, 
    does not (necessarily) represent a 1-hour maximum temperature,
    as the period over which the maximum is derived is included in the bounds
    (not shown), but rather that the temperature data used to calculate
    the maximum was provided in 1-hour steps.

        .. code-block:: python

            air_temperature: cell_methods = "time: maximum (interval: 1 hour)" ;

> The non-standardised part is included in the same way

Reviewer: This section uses both the American-English (standardized) and UK-English (standardised) spellings. I think the American spelling for "realizations" has been used throughout the metadata documentation, so can the whole document be aligned on this please?

(unless there is also a standardised part), for example:

.. code-block:: python
        
    air_temperature: cell_methods = "time: mean (time-weighted)" ;
 
If both standardised and non-standardised parts are present,

> the latter must be proceeded by ``comment:`` keyword;

Reviewer: ```suggestion
the latter must be preceded by the ``comment:`` keyword;
```

Author: Corrected

for an example of this, see the `CF Metadata Conventions`_.

Cell methods can be used to describe any simple statistical method

> that has been apply to a variable.

Reviewer: ```suggestion
that has been applied to a variable.
```

Author: Corrected.

2. Complex version, including a whole chain of processes that have been applied
   to the variable:

.. code-block:: python

    air_temperature:cell_methods = "time: maximum realization: mean area: mean (neighbourhood: square topographic) forecast_reference_time: mean (time-weighted) area: mean (recursive-filter) model: mean (model-weighted)" ;

The issue with the complex version is that only the ``time: maximum`` 
is required by any user to correctly interpret and use the variable.
The other processing steps tell you more about how it was generated
and are really acting as a substitute for provenance metadata.
This can obscure the essential statistical information,
making it harder to understand what the variable actually represents.

Use of cell methods in IMPROVER
*******************************

IMPROVER should only use cell methods to represent the **what** metadata
of the variable,
i.e. information that is required to correctly interpret the variable.
See the section on :ref:`principles-CF-conformance-label` 
in :ref:`principles-label`.

The use of the ``interval`` within the extra information in cell methods
is unhelpful and potentially confusing within IMPROVER

> and should be omitted.s

Reviewer: ```suggestion
and should be omitted.
```

the values representing the variable have been determined
or derived by the specified ``method``.

name
    Can be a dimension of the variable, a scalar coordinate variable,
    a valid `CF Standard Name`_, or the word ``area``. 

method
    Should be selected from a list:
    ``point``,
    ``maximum``,
    ``maximum_absolute_value``,
    ``median``,
    ``mid_range``,
    ``minimum``,
    ``minimum_absolute_value``,
    ``mean, mean_absolute_value``,
    ``mean_of_upper_decile``,
    ``mode``,
    ``range``,
    ``root_mean_square``,
    ``standard_derivation``,
    ``sum, sum_of_squares``,
    ``variance``.

The default interpretation for variables which do not have a ``cell_methods``
attribute depends on whether the quantity being described is extensive
(depends on the size of the cell along the axis of interest)
or intensive (no dependence on the size of the cell along the axis of interest).
The `CF Metadata Conventions`_ document uses precipitation as an example.
Precipitation rate is intensive; it would be described by
``cell_methods = "time: point"`` and requires no time bounds.
Precipitation accumulation is extensive; it would be described by

> ``cell_methods = "time: sum"`` and does required time bounds.

Author: Corrected.
